### PR TITLE
add_attributes() twig function

### PR DESCRIPTION
**Pull request by JeffTomlinson**
Usage:

```
{% set additional_attributes = {
  "foo": "bar",
  "baz": ["goobar", "foobar"],
} %}

<div {{ add_attributes(additional_attributes) }}></div>
```

- Test whether you can use the bem function inside the add_attributes function (or whether they should)

### CODE REVIEW

**Comment by ModulesUnraveled** [comment]
Does what it says on the tin.

Is it possible to use this with the BEM function from [114](https://github.com/fourkitchens/emulsify/pull/114)? Or are the mutually exclusive?

File: components/_patterns/01-atoms/02-text/00-headings/_heading.twig [hunk 1]
```diff
@@ -18,7 +18,12 @@
 #}
 {% set heading_base_class = heading_base_class|default('h' ~ heading_level) %}
 
-<h{{ heading_level }} {{ bem(heading_base_class, (heading_modifiers), heading_blockname) }}>
+{% set additional_attributes = {
+  "class": [bem(heading_base_class, (heading_modifiers), heading_blockname)],
```
Comment (ModulesUnraveled) [inline components/_patterns/01-atoms/02-text/00-headings/_heading.twig]: This is great! works for me! But I think it definitely needs to be in documentation, not actually implemented here by default.

**Comment by evanmwillhite** [comment]
Ok, this has been refactored to account for Drupal and Pattern Lab. I'm decently happy with how it's handled, but I'd like for a PHP dev to weigh in. There are some constraints around using it that will need to be documented though. Basically, they are as follows:

1. The BEM function must be written as an array

```
{% set additional_attributes = {
  "class": [bem(heading_base_class, (heading_modifiers), heading_blockname)],
} %}
```

2. Because of that, nothing else can use an array (this won't work)

```
{% set additional_attributes = {
  "class": ['red', 'blue'],
} %}
```

3. Instead, you would write that above as a string (this will work)

```
{% set additional_attributes = {
  "class": 'red blue',
} %}
```

**Comment by ModulesUnraveled** [comment]
Looks like some things have changed, so I'm removing the `passes functional review` label

**Comment by JeffTomlinson** [comment]
@evanmwillhite @ModulesUnraveled That last commit should add some flexibility so you can do things like:
```
{% set additional_attributes = {
  "class": bem(heading_base_class, (heading_modifiers), heading_blockname),
} %}
```
and:
```{% set additional_attributes = {
  "class": [bem(heading_base_class, (heading_modifiers), heading_blockname), "foo", "bar"],
} %}
```

**Comment by ModulesUnraveled** [comment]
W00t! Works great!

**Comment by evanmwillhite** [comment]
Moved over [here](https://github.com/drupal-pattern-lab/add-attributes-twig-extension). Thanks all!

**Pull request closed without merging**